src/loaders/tiff/ome-tiff.ts

```diff
@@ -0,0 +1,71 @@
+import type { GeoTIFF } from 'geotiff';
+import { fromString } from '../omexml';
+
+import TiffPixelSource from './pixel-source';
+import {
+  getLegacyIndexer,
+  getSubIFDIndexer,
+  OmeTiffIndexer
+} from './lib/indexers';
+import { getPixelSourceMeta } from './lib/utils';
+
+export interface OmeTiffSelection {
+  t: number;
+  c: number;
+  z: number;
+}
+
+export async function load(tiff: GeoTIFF) {
+  // Get first image from tiff and inspect OME-XML metadata
+  const firstImage = await tiff.getImage(0);
+  const {
+    ImageDescription,
+    SubIFDs,
+    PhotometricInterpretation: photometricInterpretation
+  } = firstImage.fileDirectory;
+  const omexml = fromString(ImageDescription);
+
+  /*
+   * Image pyramids are stored differently between versions of Bioformats.
+   * Thus we need a different indexer depending on which format we have.
+   */
+  let levels;
+  let pyramidIndexer: OmeTiffIndexer;
+
+  if (SubIFDs) {
+    // Image is >= Bioformats 6.0 and resolutions are stored using SubIFDs.
+    levels = SubIFDs.length;
+    pyramidIndexer = getSubIFDIndexer(tiff, omexml);
+  } else {
+    // Image is legacy format; resolutions are stored as separate images.
+    levels = omexml.length;
+    pyramidIndexer = getLegacyIndexer(tiff, omexml);
+  }
+
+  // TODO: The OmeTIFF loader only works for the _first_ image in the metadata.
+  const imgMeta = omexml[0];
+  const { labels, getShape, physicalSizes, dtype } = getPixelSourceMeta(
+    imgMeta
+  );
+  const tileSize = firstImage.getTileWidth();
+  const meta = { photometricInterpretation, physicalSizes };
```

Comment:
Fine with this name `meta`, but do wonder about alternatives like `renderingOpts` or something similar that have less clash with the `metadata` returned as part of the loader...I'll try to think about this

Reply:
Perhaps `pixelSourceMeta` to mirror the interface?

Reply:
This metadata is _on_ the `PixelSource`. Within the function scope, I'm okay with renaming this `pixelSourceMeta` for clarity, but feels clunky have an API like:

```javascript
const { data, metadata } = loadOmeTiff(url);

const pixelSource = data[0];
pixelSource.pixelSourceMeta
// vs
pixelSource.meta
```

If you `console.log` `data` it takes on the form:

```
// [ZarrPixelSource, ZarrPixelSource, ZarrPixelSource, ...]
```

Reply:
Right, right, ok so not that.  But maybe `renderingMetadata` - I'll try to think of something, not a big deal if I come up with nothing.  It's good as it is, just trying to avoid a situation where future people put irrelevant/unnecessary metadata into there (that has nothing to do with rendering)

Reply:
I'm going to leave it as-is for now. My hope is that people think of the pixel sources as mostly something for viv's layers and at most inspect the `shape` and `labels` properties.